### PR TITLE
fix: decouple choose crypto asset components

### PR DESCRIPTION
**Pull request by edgarkhanzadian**
> Try out this version of Leather — [download extension builds](https://github.com/leather-wallet/extension/actions/runs/7100503596).

Decouple choose crypto asset components so that those components can be later reused for choosing assets to buy. (for this ticket: https://github.com/leather-wallet/extension/issues/3188)

1st commit [8876da1](https://github.com/leather-wallet/extension/pull/4642/commits/8876da16d07a8af447156dd04a9b2e40635a1036): actual changes to decouple the components
2nd commit [029a9a1](https://github.com/leather-wallet/extension/pull/4642/commits/029a9a18f6b5d1d5c01537591432dd98c15afd33): move of the files to the components folder

### CODE REVIEW

File: src/app/pages/send/choose-crypto-asset/components/crypto-asset-list-item.tsx [hunk 1]
```diff
@@ -47,16 +20,11 @@ export function CryptoAssetListItem(props: CryptoAssetListItemProps) {
           assetBalance={assetBalance}
           icon={<CryptoCurrencyAssetIcon blockchain={blockchain} />}
           isPressable
-          onClick={() => navigateToSendForm({ isFtToken: false })}
+          onClick={props.onClick}
```
Comment (kyranjamie) [inline src/app/pages/send/choose-crypto-asset/components/crypto-asset-list-item.tsx]: Can we destructure this on L13?

File: src/app/components/crypto-assets/choose-crypto-asset/crypto-asset-list-item.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  const { blockchain, type } = assetBalance;
+
+  switch (type) {
+    case 'crypto-currency':
+      return (
+        <CryptoCurrencyAssetItem
+          assetBalance={assetBalance}
+          icon={<CryptoCurrencyAssetIcon blockchain={blockchain} />}
+          isPressable
+          onClick={props.onClick}
```
Comment (kyranjamie) [inline src/app/components/crypto-assets/choose-crypto-asset/crypto-asset-list-item.tsx]: Can we destructure on L13?

File: src/app/components/crypto-assets/choose-crypto-asset/crypto-asset-list-item.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import type { AllTransferableCryptoAssetBalances } from '@shared/models/crypto-asset-balance.model';
+
+import { CryptoCurrencyAssetItem } from '@app/components/crypto-assets/crypto-currency-asset/crypto-currency-asset-item';
+
+import { CryptoCurrencyAssetIcon } from './crypto-currency-asset-icon';
+import { FungibleTokenAssetItem } from './fungible-token-asset-item';
+
+interface CryptoAssetListItemProps {
+  assetBalance: AllTransferableCryptoAssetBalances;
+  onClick: () => void;
```
Comment (kyranjamie) [inline src/app/components/crypto-assets/choose-crypto-asset/crypto-asset-list-item.tsx]:
```suggestion
  onClick(): void;
```

this makes no material difference in the type, but would be nice to enforce a consistent way of defining fn types. Is there an eslint rule for this?

File: src/app/pages/send/choose-crypto-asset/choose-crypto-asset.tsx [hunk 4]
```diff
@@ -34,9 +41,33 @@ export function ChooseCryptoAsset() {
 
   useRouteHeader(<ModalHeader hideActions defaultGoBack title=" " />);
 
+  function navigateToSendForm({
+    cryptoAssetBalance,
+  }: {
+    cryptoAssetBalance: AllTransferableCryptoAssetBalances;
+  }) {
```
Comment (kyranjamie) [inline src/app/pages/send/choose-crypto-asset/choose-crypto-asset.tsx]: Any reason for this to be a obj rather than single prop?

**Review by kyranjamie — APPROVED** [review]
LGTM, good refactor. 

Only comments are formatting/style related

File: src/app/components/crypto-assets/choose-crypto-asset/crypto-asset-list-item.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+}
+export function CryptoAssetListItem(props: CryptoAssetListItemProps) {
+  const { assetBalance } = props;
```
Comment (pete-watters) [inline src/app/components/crypto-assets/choose-crypto-asset/crypto-asset-list-item.tsx]: Maybe we can de-structure `onClick` here also?

**Review by pete-watters — APPROVED** [review]

File: src/app/pages/send/choose-crypto-asset/choose-crypto-asset.tsx (hunk 4 above)
Comment (edgarkhanzadian) [reply src/app/pages/send/choose-crypto-asset/choose-crypto-asset.tsx]: imho using objects for arguments give function calls more readability, but i can change it to a single prop

File: src/app/components/crypto-assets/choose-crypto-asset/crypto-asset-list-item.tsx (hunk 3 above)
Comment (kyranjamie) [reply src/app/components/crypto-assets/choose-crypto-asset/crypto-asset-list-item.tsx]: there is! https://typescript-eslint.io/rules/method-signature-style/#method

Maybe we can add this rule to prefer methods here?

Comment (edgarkhanzadian) [reply src/app/components/crypto-assets/choose-crypto-asset/crypto-asset-list-item.tsx]: Just found it too! :D yea will add it now

**Comment by edgarkhanzadian** [comment]
thank you @pete-watters @kyranjamie for the reviewes!

**Pull request merged**